internal/controlplane/handlers_evalstatus.go

```diff
@@ -17,220 +17,324 @@ package controlplane
 
 import (
 	"context"
-	"time"
+	"fmt"
 
+	"github.com/google/uuid"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
-	"github.com/stacklok/minder/internal/engine"
+	"github.com/stacklok/minder/internal/db"
 	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
 )
 
-// ListEvaluationResults lists the evaluation results for entities filtered by entity type, labels, profiles, and rule types.
+// ListEvaluationResults lists the evaluation results for entities filtered b
+// entity type, labels, profiles, and rule types.
 func (s *Server) ListEvaluationResults(
 	ctx context.Context,
-	_ *minderv1.ListEvaluationResultsRequest,
+	in *minderv1.ListEvaluationResultsRequest,
 ) (*minderv1.ListEvaluationResultsResponse, error) {
-	entityCtx := engine.EntityFromContext(ctx)
+	if _, err := uuid.Parse(in.GetProfile()); err != nil && in.GetProfile() != "" {
+		return nil, status.Error(codes.InvalidArgument, "Error invalid profile ID")
+	}
 
-	err := entityCtx.Validate(ctx, s.store)
+	projectID, err := uuid.Parse(*in.GetContext().Project)
 	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "error in entity context: %v", err)
+		return nil, fmt.Errorf("error parsing project identifier")
```

Comment:
why not just use:

```
	entityCtx := engine.EntityFromContext(ctx)
	projectID := entityCtx.Project.ID
```

??

Reply:
I was trying to validate the input but it's already checked elsewhere I can drop this check :)